test/integration/datapath/datapath_linux_test.go

```diff
@@ -91,103 +91,85 @@ func setupLinuxEnvironment(t *testing.T) {
-		deployment.Spec.Template.ObjectMeta.Labels[podLabelKey] = *podPrefix
-		deployment.Spec.Template.Spec.NodeSelector[nodepoolKey] = *nodepoolSelector
-		deployment.Name = *podPrefix
-		deployment.Namespace = *podNamespace
-
-		deploymentsClient := clientset.AppsV1().Deployments(*podNamespace)
-		err = k8sutils.MustCreateDeployment(ctx, deploymentsClient, deployment)
+	} else {
+		deployment, err = k8sutils.MustParseDeployment(LinuxDeployIPV4)
 		if err != nil {
 			require.NoError(t, err)
 		}
 
-		daemonsetClient := clientset.AppsV1().DaemonSets(daemonset.Namespace)
-		err = k8sutils.MustCreateDaemonset(ctx, daemonsetClient, daemonset)
+		daemonset, err = k8sutils.MustParseDaemonSet(gpDaemonset)
 		if err != nil {
 			t.Fatal(err)
 		}
+	}
+
+	// setup common RBAC, ClusteerRole, ClusterRoleBinding, ServiceAccount
+	rbacSetupFn, err := k8sutils.MustSetUpClusterRBAC(ctx, clientset, gpClusterRolePath, gpClusterRoleBindingPath, gpServiceAccountPath)
+	if err != nil {
+		t.Log(os.Getwd())
+		t.Fatal(err)
+	}
 
-		t.Cleanup(func() {
-			t.Log("cleaning up resources")
-			rbacSetupFn()
+	// Fields for overwritting existing deployment yaml.
+	// Defaults from flags will not change anything
+	deployment.Spec.Selector.MatchLabels[podLabelKey] = *podPrefix
+	deployment.Spec.Template.ObjectMeta.Labels[podLabelKey] = *podPrefix
+	deployment.Spec.Template.Spec.NodeSelector[nodepoolKey] = *nodepoolSelector
+	deployment.Name = *podPrefix
+	deployment.Namespace = *podNamespace
+	daemonset.Namespace = *podNamespace
+
+	deploymentsClient := clientset.AppsV1().Deployments(*podNamespace)
+	err = k8sutils.MustCreateDeployment(ctx, deploymentsClient, deployment)
+	if err != nil {
+		require.NoError(t, err)
+	}
 
-			if err := deploymentsClient.Delete(ctx, deployment.Name, metav1.DeleteOptions{}); err != nil {
-				t.Log(err)
-			}
+	daemonsetClient := clientset.AppsV1().DaemonSets(daemonset.Namespace)
+	err = k8sutils.MustCreateDaemonset(ctx, daemonsetClient, daemonset)
+	if err != nil {
+		t.Fatal(err)
+	}
 
-			if err := daemonsetClient.Delete(ctx, daemonset.Name, metav1.DeleteOptions{}); err != nil {
-				t.Log(err)
-			}
-		})
+	t.Cleanup(func() {
+		t.Log("cleaning up resources")
+		rbacSetupFn()
 
-		t.Log("Waiting for pods to be running state")
-		err = k8sutils.WaitForPodsRunning(ctx, clientset, *podNamespace, podLabelSelector)
-		if err != nil {
-			require.NoError(t, err)
+		if err := deploymentsClient.Delete(ctx, deployment.Name, metav1.DeleteOptions{}); err != nil {
+			t.Log(err)
 		}
 
-		if *isDualStack {
-			t.Log("Successfully created customer dualstack Linux pods")
-		} else {
-			t.Log("Successfully created customer singlestack Linux pods")
+		if err := daemonsetClient.Delete(ctx, daemonset.Name, metav1.DeleteOptions{}); err != nil {
+			t.Log(err)
 		}
+	})
+
+	t.Log("Waiting for pods to be running state")
+	err = k8sutils.WaitForPodsRunning(ctx, clientset, *podNamespace, podLabelSelector)
+	if err != nil {
+		require.NoError(t, err)
+	}
```

Comment:
As this is part of setting up the test environment, shouldn't this call t.fatal()?